fix: [SNOW-3481851] skip main_file auto-insert when covered by an existing artifact (#2971)
Snowflake-Labs/snowcli: RELEASE-NOTES.md

* Added `--force` flag to `snow spcs service drop` to allow dropping services that contain block storage volumes.

## Fixes and improvements

> * Fixed `snow streamlit deploy` skipping directory artifacts when `main_file` is already included by an artifact directory.

Reviewer: What do you think about such wording? Users affected by this bug are more likely searching for TooManyFilesError which is what they see in console. Also skipping might be slightly misleading as ultimately at file gets uploaded
```suggestion
* Fixed `TooManyFilesError` during `snow streamlit deploy` when `main_file` is a descendant of a directory listed in `artifacts`.
```

Author: Thanks for the feedback - this makes sense.
